### PR TITLE
Integrate Makefile into automation

### PR DESCRIPTION
**Pull request by glower**
* Integrated Makefile into automation processes
* Updated documentation about Makefile usage
* Added documentation about LXD build
* Added missing Makefile targets for frontend tests and for code vendoring

### CODE REVIEW

**Comment by glower** [comment]
I added two new github actions to run frontend tests and backend tests. Currently we use TravisCI for this tasks. I would suggest to deactivate TravisCI integration. In my experience github actions are much faster and more reliable.

File: run-checks
Comment (glower) [inline run-checks:169]: we are building in a separate action, so don't need to do this here

File: README.md
Comment (doismellburning) [inline README.md:35]: s/build/built/

File: README.md
Comment (doismellburning) [inline README.md:61]: s/need/need a/

File: README.md
Comment (doismellburning) [inline README.md:62]: s/with/with the/

File: README.md
Comment (doismellburning) [inline README.md:146]: s/runns/runs/

File: README.md
Comment (doismellburning) [inline README.md:36]: s/with/wish/

File: README.md
Comment (doismellburning) [inline README.md:37]: s/container/container/

**Review by doismellburning — CHANGES_REQUESTED** [review]
There is quite a lot going on here that makes it more challenging to meaningfully review than I'd like; could you try to keep it to commit-per-logical-change please? Trying to follow what you're trying to do is hard when e.g. the README diff has everything from formatting changes to adding LXD instructions to moving to Make usage etc.

Splitting it up would not only make it easier for me to review, it will help when looking back at history, trying to understand intent, etc. - I'd appreciate it if you would please.

**Comment by glower** [comment]
> Splitting it up would not only make it easier for me to review, it will help when looking back at history, trying to understand intent, etc. - I'd appreciate it if you would please.

Sure, make sense. I will split the automation changes and README changes

File: run-checks
Comment (glower) [inline run-checks:16]: '-mod=vendor' tells go to use vendor code from the `vendor` folder

File: .github/workflows/nodejs.yml
Comment (glower) [inline .github/workflows/nodejs.yml:17]: this is moved to the Makefile

File: .travis.yml
Comment (glower) [inline .travis.yml:1]: this is all not needed, new config file is clean

File: .travis.yml
Comment (glower) [inline .travis.yml:50]: codecov never worked for the frontend tests, removed

**Comment by glower** [comment]
* New github actions are removed and will be introduced in the new PR
* Unnecessary changes in the README are removed
* Added missing Makefile targets for frontend tests and for code vendoring

Documentation about LXD container could me also moved to a separate PR but I think it make sense here too

File: .travis.yml
Comment (Roadmaster) [inline .travis.yml:1]: Note that it'd be best to test on the same distro where the service is deployed. That said, it's OK to leave this as focal and we should instead change the build/deploy process to use focal :) (tl;dr no change needed here).

**Review by Roadmaster — APPROVED** [review]
LGTM, a tiny comment but should not block the merge.

I agree with Kristian's request to split this into meaningful commits but if the resulting diff is the same, shouldn't need to request another review.

File: .travis.yml
Comment (glower) [reply .travis.yml:1]: Yes, that makes sense. I will change the build/test process to use focal in the next PR

**Pull request merged**
